src/intrinsics/prepack/global.js

```diff
@@ -227,8 +227,11 @@ export default function(realm: Realm): void {
       "__evaluatePureFunction",
       0,
       (context, [functionValue]) => {
-        invariant(functionValue instanceof ECMAScriptSourceFunctionValue);
-        invariant(typeof functionValue.$Call === "function");
+        if (!(functionValue instanceof ECMAScriptSourceFunctionValue) || typeof functionValue.$Call !== "function") {
+          let error = new CompilerDiagnostic("CompilerDiagnostic Error");
```

Comment:
Please have a look at other places where CompilerDiagnostic instances are created. There are niceties to be observed and you should not throw diagnostics.